### PR TITLE
2.6.x Spring Event & Bugfix

### PR DESCRIPTION
**Pull request by mercyblitz**
## New Features

* [Introduce a new Spring ApplicationEvent for ServiceBean exporting](https://github.com/apache/incubator-dubbo/issues/2251)
   *  Refer to `com.alibaba.dubbo.config.spring.context.event.ServiceBeanExportedEvent`

## Enhancement

* [Reactor the generation rule for @Service Bean name](https://github.com/apache/incubator-dubbo/issues/2235)
  * Refer to `com.alibaba.dubbo.config.spring.beans.factory.annotation.ServiceBeanNameBuilder`

## Fixed Bugs

* [@Reference field can't refer its' @Service Bean in same JVM](https://github.com/apache/incubator-dubbo/issues/1654)
   * Re-run Test case `ReferenceAnnotationBeanPostProcessorTest`

### CODE REVIEW

**Comment by mercyblitz** [comment]
Please focus on code review since those commits have some break update.

File: dubbo-config/dubbo-config-spring/src/main/java/com/alibaba/dubbo/config/spring/ServiceBean.java [hunk 1]
```diff
@@ -16,23 +16,16 @@
...
 package com.alibaba.dubbo.config.spring;
 
-import com.alibaba.dubbo.config.ApplicationConfig;
-import com.alibaba.dubbo.config.ModuleConfig;
-import com.alibaba.dubbo.config.MonitorConfig;
-import com.alibaba.dubbo.config.ProtocolConfig;
-import com.alibaba.dubbo.config.ProviderConfig;
-import com.alibaba.dubbo.config.RegistryConfig;
-import com.alibaba.dubbo.config.ServiceConfig;
+import com.alibaba.dubbo.config.*;
```
Comment (diecui1202) [inline dubbo-config/dubbo-config-spring/src/main/java/com/alibaba/dubbo/config/spring/ServiceBean.java]: make sure not add import *

Comment (mercyblitz) [reply dubbo-config/dubbo-config-spring/src/main/java/com/alibaba/dubbo/config/spring/ServiceBean.java]: Ok, it requires change IDEA Template

Comment (zonghaishang) [reply dubbo-config/dubbo-config-spring/src/main/java/com/alibaba/dubbo/config/spring/ServiceBean.java]: Code style template: https://github.com/apache/incubator-dubbo/blob/master/codestyle/dubbo_codestyle_for_idea.xml

**Comment by diecui1202** [comment]
@mercyblitz Pls. check the ci failure.

File: dubbo-config/dubbo-config-spring/src/main/java/com/alibaba/dubbo/config/spring/beans/factory/annotation/ServiceBeanNameBuilder.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+import com.alibaba.dubbo.config.spring.ServiceBean;
+import org.springframework.core.env.Environment;
+import org.springframework.util.StringUtils;
+
+import static com.alibaba.dubbo.config.spring.util.AnnotationUtils.resolveInterfaceName;
+
+/**
+ * Dubbo {@link Service @Service} Bean Builder
+ *
+ * @author <a href="mailto:mercyblitz@gmail.com">Mercy</a>
```
[the remark this answers is not in the document]
Comment (mercyblitz) [reply dubbo-config/dubbo-config-spring/src/main/java/com/alibaba/dubbo/config/spring/beans/factory/annotation/ServiceBeanNameBuilder.java]: Ok

**Pull request closed without merging**